bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/discovery/SmartHomeDevicesDiscovery.java

```diff
@@ -0,0 +1,255 @@
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonSmartHomeGroups.SmartHomeGroup;
+import org.openhab.binding.amazonechocontrol.internal.jsons.SmartHomeBaseDevice;
+import org.openhab.binding.amazonechocontrol.internal.smarthome.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Lukas Knoeller - Initial contribution
+ */
+@NonNullByDefault
+public class SmartHomeDevicesDiscovery extends AbstractDiscoveryService implements ExtendedDiscoveryService {
+
+    private AccountHandler accountHandler;
+    private final Logger logger = LoggerFactory.getLogger(SmartHomeDevicesDiscovery.class);
+
+    private @Nullable ScheduledFuture<?> startScanStateJob;
+    private @Nullable Long activateTimeStamp;
+
+    private @Nullable DiscoveryServiceCallback discoveryServiceCallback;
+
+    @Override
+    public void setDiscoveryServiceCallback(DiscoveryServiceCallback discoveryServiceCallback) {
+        this.discoveryServiceCallback = discoveryServiceCallback;
+    }
+
+    public SmartHomeDevicesDiscovery(AccountHandler accountHandler) {
+        super(SUPPORTED_SMART_HOME_THING_TYPES_UIDS, 10);
+        this.accountHandler = accountHandler;
+    }
+
+    public void activate() {
+        activate(new Hashtable<String, @Nullable Object>());
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    protected void startScan() {
+        stopScanJob();
+        Long activateTimeStamp = this.activateTimeStamp;
+        if (activateTimeStamp != null) {
+            removeOlderResults(activateTimeStamp);
+        }
+        setSmartHomeDevices(accountHandler.updateSmartHomeDeviceList(false));
+    }
+
+    protected void startAutomaticScan() {
+        if (!this.accountHandler.getThing().getThings().isEmpty()) {
+            stopScanJob();
+            return;
+        }
+        Connection connection = this.accountHandler.findConnection();
+        if (connection == null) {
+            return;
+        }
+        Date verifyTime = connection.tryGetVerifyTime();
+        if (verifyTime == null) {
+            return;
+        }
+        if (new Date().getTime() - verifyTime.getTime() < 10000) {
+            return;
+        }
+        startScan();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        stopScanJob();
+        startScanStateJob = scheduler.scheduleWithFixedDelay(this::startAutomaticScan, 3000, 1000,
+                TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        stopScanJob();
+    }
+
+    void stopScanJob() {
+        ScheduledFuture<?> currentStartScanStateJob = startScanStateJob;
+        if (currentStartScanStateJob != null) {
+            currentStartScanStateJob.cancel(false);
+            startScanStateJob = null;
+        }
+        super.stopScan();
+    }
+
+    @Override
+    @Activate
```

Comment:
Since this class isn't an osgi service I don't know if adding `@Activate` serves any purpose here.

Reply:
Not sure either. It's the same in the "old" `AmazonEchoDiscovery`. Since it does no harm, I would suggest to leave  it here (since the discovery needs to be rewritten anyway).